core/src/main/scala/org/apache/spark/network/netty/NettyBlockTransferService.scala

```diff
@@ -48,6 +54,41 @@ class NettyBlockTransferService(conf: SparkConf, securityManager: SecurityManage
   private[this] var clientFactory: TransportClientFactory = _
   private[this] var appId: String = _
 
+  override def metricsCollection(executorId: String, executorMetrics: ExecutorMetrics): Unit = {
+    logDebug("starting a new thread for metricsCollection()")
+    new Thread("metric thread for executor " + executorId) {
+      override def run() {
+        while (true) {
+          getNettyMemMetrics(executorMetrics)
+          Thread.sleep(2000)
```

Comment:
Yes, I just want a finer resolution, it is discussed in the design doc, but I think you are right, we shall not create a thread for just one metric. I'll get the metric in each heartbeat for current implementation.